### PR TITLE
fix(ScioContext): Use the given parameter of setJobName

### PR DESCRIPTION
**Pull request by aseure**
cc @ElPicador

### CODE REVIEW

File: scio-core/src/main/scala/com/spotify/scio/ScioContext.scala [hunk 1]
```diff
@@ -305,8 +305,13 @@ class ScioContext private[scio] (val options: PipelineOptions,
     if (_pipeline != null) {
       throw new RuntimeException("Cannot set job name once pipeline is initialized")
     }
-    Try(optionsAs[DataflowPipelineOptions])
-      .foreach(_.setJobName(new DataflowPipelineOptions.JobNameFactory().create(options)))
+    Try(optionsAs[DataflowPipelineOptions]).foreach { opts =>
+      if (name != "") {
```
Comment (ravwojdyla) [inline scio-core/src/main/scala/com/spotify/scio/ScioContext.scala]: How about just passing the name, and delegating the checks to the sdk (which happens locally, before the job is submitted)?

File: scio-test/src/test/scala/com/spotify/scio/ScioContextTest.scala [hunk 2]
```diff
@@ -67,14 +67,23 @@ class ScioContextTest extends PipelineSpec {
...
 
+  it should "support user defined job name via context" in {
+    val jobName = "test-job-1"
+    val opts = PipelineOptionsFactory.create().as(classOf[DataflowPipelineOptions])
+    val sc = ScioContext(opts)
+    sc.setJobName(jobName)
+    val pipelineOpts = sc.pipeline.getOptions.as(classOf[DataflowPipelineOptions])
+    pipelineOpts.getJobName shouldBe jobName
+  }
+
```
Comment (ravwojdyla) [inline scio-test/src/test/scala/com/spotify/scio/ScioContextTest.scala]: Could you please also add test where:
 * job name is set via options
 * followed by set (with different name) via context
and the end name should be the one provided via  context.

File: scio-core/src/main/scala/com/spotify/scio/ScioContext.scala (hunk 1 above)
Comment (aseure) [inline scio-core/src/main/scala/com/spotify/scio/ScioContext.scala]: Yes indeed. I'll do it.

File: scio-test/src/test/scala/com/spotify/scio/ScioContextTest.scala (hunk 2 above)
Comment (aseure) [inline scio-test/src/test/scala/com/spotify/scio/ScioContextTest.scala]: Will do 👍

**Comment by ravwojdyla** [comment]
Looks good, thanks and good find @aseure !

**Comment by aseure** [comment]
@ravwojdyla When do you plan to create a new `scio` release with this fix?

**Comment by ravwojdyla** [comment]
@aseure `0.2.11` was "just" released, comms/release page will be updated soon.

**Pull request merged**